util/binaryserializer/binaryserializer.go

```diff
@@ -113,19 +113,21 @@ func PutUint16(w io.Writer, val uint16) error {
 // buffer from the free list and writes the resulting four bytes to the given
 // writer.
 func PutUint32(w io.Writer, val uint32) error {
-	var buf [4]byte
-	binary.LittleEndian.PutUint32(buf[:], val)
-	_, err := w.Write(buf[:])
+	buf := Borrow()[:4]
```

Comment:
Can you explain how does it leak memory? `buf` is an array, so it should be stored entirely on the stack. I wouldn't expect it to leak.

Reply:
And also, `Borrow` is pretty discouraged because it uses a `select` which can be quite slow. It's better to get rid of it where possible.